### PR TITLE
New domain for RN Docs: reactnative.dev

### PR DESCRIPTION
**Pull request by rachelnabors**
Updates links, config, and such to work with react.native.dev, shifting us off the old *facebook.github.io/react-native/* URL scheme

### CODE REVIEW

File: docs/network.md [hunk 1]
```diff
@@ -59,11 +59,17 @@ You can also use the `async` / `await` syntax in a React Native app:
 ```jsx
 async function getMoviesFromApiAsync() {
   try {
+<<<<<<< HEAD
```
Comment (Simek) [inline docs/network.md]: There are Git merge chunks committed in `docs/network.md`.

**Comment by Simek** [comment]
Also some of `.md` extensions were removed from the links but there are still 250+ occurrences in the `master` docs. Is it desirable to remove all of them?

**Comment by Simek** [comment]
Repository description also needs an update. But unfortunately currently I cannot access website by  https://reactnative.dev/ URL due to SSL certificate signing/configuration issue.

**Comment by rachelnabors** [comment]
> Also some of .md extensions were removed from the links but there are still 250+ occurrences in the master docs on this branch. Is it desirable to remove all of them?

Turned out there was no real benefit to using or not using them that I could see. Leave as is unless something indicates otherwise.

**Pull request merged**